### PR TITLE
Added "timestamp-trigger" Profile

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added Timestamp on Trigger Profile

This Profile updates the State of an DateTime Item to now each time the Channel is triggered.

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/main/resources/OH-INF/i18n/SystemProfiles.properties [hunk 1]
```diff
@@ -23,4 +23,5 @@ profile.config.system.timestamp-offset.offset.label = Offset
 profile.config.system.timestamp-offset.offset.description = Offset to be applied on the state towards the item. The negative offset will be applied in the reverse direction.
 profile.config.system.timestamp-offset.timezone.label = Time Zone
 profile.config.system.timestamp-offset.timezone.description = A time zone to be applied on the state.
+profile-type.system.timestamp-trigger.label = Timestamp on Trigger
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.thing/src/main/resources/OH-INF/i18n/SystemProfiles.properties:26]:
```suggestion
profile-type.system.timestamp-trigger.label = Timestamp on trigger
```

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemProfileFactory.java [hunk 2]
```diff
@@ -179,7 +181,7 @@ public SystemProfileFactory(final @Reference ChannelTypeRegistry channelTypeRegi
                 case STATE:
                     return DEFAULT;
                 case TRIGGER:
-                    return null;
+                    return TIMESTAMP_TRIGGER;
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemProfileFactory.java:184]: Pfew, I am not so sure if this profile should really become the generally suggested one...
I was just [looking into profiles](https://github.com/openhab/openhab-webui/issues/886#issuecomment-844381424) today and was thinking that a "map-to-string" profile (need to find a better name) would probably make sense as a default profile for trigger channels. Wdyt?

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Many thanks! Just two small comments/questions.

File: bundles/org.openhab.core.thing/src/main/resources/OH-INF/i18n/SystemProfiles.properties (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/resources/OH-INF/i18n/SystemProfiles.properties:26]: We changed spelling for other timestamp Profiles to camel case in https://github.com/openhab/openhab-core/pull/2353 (e.g. "Timestamp on Change"). I would like to keep it.

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemProfileFactory.java (hunk 2 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemProfileFactory.java]: I reverted this change.

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemProfileFactory.java]: If I understand your idea correctly you like to add a `TriggerProfile` which converts the triggered event into a `StringType`? 

What about simply "toString"?

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**